src/header.rs

```diff
@@ -94,15 +94,31 @@ pub struct Header {
 
     /// Row order of the image data within the file
     pub row_order: RowOrder,
+
+    /// Color table for color mapped 1bpp images.
+    pub color_table: Option<[u32; 2]>,
 }
 
+/// Basic file header size in bytes. The Microsoft documentation calls this `tagBITMAPFILEHEADER`.
+const FILE_HEADER_SIZE: usize = 14;
+
+/// Size of BITMAPV4HEADER in bytes. Note that this does not include the common fields counted in
+/// [`FILE_HEADER_SIZE`].
+///
+/// Additional header sizes can be found
+/// [here](https://www.liquisearch.com/bmp_file_format/file_structure/dib_header_bitmap_information_header).
+const V4_HEADER_SIZE: usize = 108;
+
 impl Header {
     pub(crate) fn parse(input: &[u8]) -> IResult<&[u8], Header> {
+        let orig = input;
```

Comment:
Is there a way to use `input` instead of keeping a pointer to the start of the file here?